amy/extcomments/tests.py

```diff
@@ -1 +1,39 @@
-# Create your tests here.
+from django.test import TestCase
+import django_comments
+
+from workshops.models import Organization, Person
+
+
+class TestEmailFieldRequiredness(TestCase):
+    def test_email_field_requiredness(self):
+        """Regression test for #1944.
+
+        Previously a user without email address would not be able to add a comment."""
+        # Arrange
+        person = Person.objects.create(
+            personal="Ron",
+            family="Weasley",
+            username="rw",
+            is_active=True,
+            email="",
```

Comment:
This is an interesting topic. `Person.email` is marked with `unique=True`, and additionally we know it's a string field (`CharField`).

The uniqueness is only enforced on the string values of the field. This means that if we have a case in AMY where we want to enter users with not known email, we should enter `email=None` because otherwise we'd receive Integrity Error (unique constraint failed). Additionally, this also means that there can be at most one user with `email=""`.

The test should work with both `email=""` and `email=None`, as the crucial part is not having to provide any email in the form's data.